go/main.go

```diff
@@ -15,25 +15,72 @@
 package main
 
 import (
-	"fmt"
 	"log"
 	"net/http"
 	"os"
+
+	"github.com/gin-contrib/multitemplate"
+	"github.com/gin-gonic/gin"
 )
 
+type templateArgs struct {
+	MainID           string
+	Title            string
+	SubPageTitle     string
+	GA               string
+	IsHideFullFooter bool
+	HideSearch       bool
+}
+
+var urlMap = URLMap{
+	Home:       "/",
+	Place:      "/place",
+	Browser:    "/browser",
+	Timeline:   "/tools/timeline",
+	About:      "/about",
+	Dataset:    "/dataset",
+	Faq:        "/faq",
+	Feedback:   "/feedback",
+	Search:     "/search",
+	Disclaimer: "/disclaimer",
+}
+
 func main() {
-	http.HandleFunc("/go/", handle)
 	port := os.Getenv("PORT")
 	if port == "" {
-		port = "8080"
+		port = ":7070"
 	}
 	log.Printf("Listening on port %s", port)
-	if err := http.ListenAndServe(":"+port, nil); err != nil {
-		log.Fatal(err)
-	}
-}
 
-func handle(w http.ResponseWriter, r *http.Request) {
-	log.Println(r.URL.Path)
-	fmt.Fprint(w, "Hello Data Commons!")
+	r := gin.Default()
+
+	r.Static("/static", "./dist")
+
+	// Load template HTML files.
+	r.LoadHTMLGlob("template/*/*")
+
+	tmpl := multitemplate.New()
+	tmpl.AddFromFiles("dev", "template/base.html", "template/dev/dev.html")
+	r.HTMLRender = tmpl
+
+	// TODO(shifucun): still keep the /dev page until the webdriver tests are
+	// setup for the Go service.
+	r.GET("/go/dev", func(ctx *gin.Context) {
+		if os.Getenv("FLASK_ENV") == "production" {
+			ctx.JSON(404, gin.H{"message": "Page not found"})
+		} else {
+			ctx.HTML(http.StatusOK, "dev", gin.H{
+				"data": templateArgs{
+					MainID:           "dev",
```

Comment:
is it not possible to define these in the template anymore?

Reply:
Did some research and don't see a way to share variable among the golang html template. The golang text/html template does not have many magic. It's a rather straightforward system that takes variable and function and replace them in the template.

A good thing to explicitly list the arguments is to have all of them in one file for easy maintenance.